Applications/Spire/Source/OrderImbalanceIndicator/CurrentOrderImbalanceIndicatorTableModel.cpp

```diff
@@ -0,0 +1,139 @@
+#include "Spire/OrderImbalanceIndicator/CurrentOrderImbalanceIndicatorTableModel.hpp"
+#include <Beam/Queues/Publisher.hpp>
+#include <Beam/Threading/Timer.hpp>
+
+using namespace Beam;
+using namespace Beam::Threading;
+using namespace Beam::TimeService;
+using namespace boost;
+using namespace boost::gregorian;
+using namespace boost::posix_time;
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+CurrentOrderImbalanceIndicatorTableModel::CallbackTimer::CallbackTimer(
+    std::unique_ptr<TimerBox> timer, std::function<void ()> expiration_callback)
+    : m_timer(std::move(timer)),
+      m_expiration_callback(std::move(expiration_callback)) {
+  m_timer->GetPublisher().Monitor(
+    m_queue.GetSlot<Timer::Result>([=] (auto result) {
+      if(result == Timer::Result::EXPIRED) { m_expiration_callback(); }
+    }));
+  m_timer->Start();
+}
+
+CurrentOrderImbalanceIndicatorTableModel::CurrentOrderImbalanceIndicatorTableModel(
+    time_duration offset, TimeClientBox clock, TimerFactory timer_factory,
+    std::shared_ptr<OrderImbalanceIndicatorModel> source)
+    : m_source(std::move(source)),
```

Comment:
The ordering is inconsistent here. Order things as you would according to English grammar and use a consistent ordering.

Reply:
Are you referring to the ordering of the parameters in the API in contrast to the ordering as they're declared in the class? I'm just not sure what that has to do with grammar.

Reply:
The whole thing, use a consistent ordering according to how most English speakers speak and keep that ordering consistent. Ordering matters, it's not arbitrary and it informs things such as dependencies and relationships. For example it's why almost all programming languages structure things as ```subject.verb(noun1, noun2, ...)``` or ```verb(noun1, noun2, ...)```.